crypto/merkle.go

```diff
@@ -124,10 +124,12 @@ func MerkleProof(b []byte, proofIndex uint64) (base []byte, hashSet []Hash) {
 
 	// Get the proof and convert it to a base + hash set.
 	_, proof, _, _ := t.Prove()
-	base = proof[0]
-	hashSet = make([]Hash, len(proof)-1)
-	for i, p := range proof[1:] {
-		copy(hashSet[i][:], p)
+	if len(proof) > 0 {
```

Comment:
since the code relies on `proof[0]` not panicking, I think it makes sense to check `len(proof)` rather than `len(b)`.